### ISSUE TITLE
[Bug]: VllmServer returns "Exception: can only concatenate str (not "NoneType") to str" with stream_complete

### ISSUE THREAD

**Issue by tonyaw** [run-llama/llama_index#11029]
### Bug Description

I'm trying to use private_gpt to call VllmServer, and got following backtrace:
```
chunk=b''
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/queueing.py", line 459, in call_prediction
    output = await route_utils.call_process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/blocks.py", line 1533, in process_api
    result = await self.call_function(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/blocks.py", line 1163, in call_function
    prediction = await utils.async_iteration(iterator)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/utils.py", line 519, in async_iteration
    return await iterator.__anext__()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/utils.py", line 640, in asyncgen_wrapper
    async for response in f(*args, **kwargs):
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/chat_interface.py", line 481, in _stream_fn
    first_response = await async_iteration(generator)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/utils.py", line 519, in async_iteration
    return await iterator.__anext__()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/utils.py", line 512, in __anext__
    return await anyio.to_thread.run_sync(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/utils.py", line 495, in run_sync_iterator_async
    return next(iterator)
           ^^^^^^^^^^^^^^
  File "/home/worker/app/private_gpt/ui/ui.py", line 278, in _chat
    yield from yield_deltas(query_stream)
  File "/home/worker/app/private_gpt/ui/ui.py", line 220, in yield_deltas
    for delta in stream:
  File "/home/worker/app/.venv/lib/python3.11/site-packages/llama_index/core/chat_engine/types.py", line 173, in response_gen
    self._unformatted_response += delta
TypeError: can only concatenate str (not "NoneType") to str

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/queueing.py", line 497, in process_events
    response = await self.call_prediction(awake_events, batch)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/.venv/lib/python3.11/site-packages/gradio/queueing.py", line 468, in call_prediction
    raise Exception(str(error) if show_error else None) from error
Exception: can only concatenate str (not "NoneType") to str
```
I print chunk in stream_complete, and it looks like this error happens when chunk is empty string(b'').
Any idea about how to fix this issue?
```python
        def gen() -> CompletionResponseGen:
            for chunk in response.iter_lines(
                chunk_size=8192, decode_unicode=False, delimiter=b"\0"
            ):
                print(f"chunk={chunk}") # <<<< print is here.
                if chunk:
                    data = json.loads(chunk.decode("utf-8"))
                    yield CompletionResponse(text=data["text"][0])
```

### Version

llama_index-0.10.6 llama_index_llms_vllm-0.1.2

### Steps to Reproduce

Send response with empty string(b'') to VllmServer.stream_complete

**Comment by logan-markewich** [comment]
This requires a PR and version bump to fix (but also, I wonder why the chunk is None, that's kind of annoying)

**Comment by logan-markewich** [comment]
Fixed in `llama-index-llms-vllm=0.1.4`

**Comment by tonyaw** [comment]
It works. Thanks for your support very much!
